### ISSUE TITLE
Visualize/Draw google's quickdraw data

### ISSUE THREAD

**Issue by shubhank008** [cduck/drawSvg#5]
I wanted to try and use Google Quick Draw's stroke data to draw and visualize the drawing like the do on their preview page but could not get to use their stroke data since I think DrawSVG and QD uses slightly different format.  

QuickDraw - https://github.com/googlecreativelab/quickdraw-dataset
Preview - https://quickdraw.withgoogle.com/data

Quick Draw provides strokes data for their drawing like this

Example strokes for a drawing
`[[[251,255,242,232,237,249],[0,153,153,166,170,160]],[[253,253,248,236,232],[154,164,172,174,167]],[[2,9,32,45,59,69,66],[2,32,88,102,107,103,94]],[[0,6,11,31,58,79,86,86,69],[3,19,73,111,128,128,122,106,91]]]`

From their page:

```
The format of the drawing array is as following:

[ 
  [  // First stroke 
    [x0, x1, x2, x3, ...],
    [y0, y1, y2, y3, ...],
    [t0, t1, t2, t3, ...]
  ],
  [  // Second stroke
    [x0, x1, x2, x3, ...],
    [y0, y1, y2, y3, ...],
    [t0, t1, t2, t3, ...]
  ],
  ... // Additional strokes
]
Where x and y are the pixel coordinates, and t is the time in milliseconds since the first point. x and y are real-valued while t is an integer. The raw drawings can have vastly different bounding boxes and number of points due to the different devices used for display and input.
```

Would appreciate if you could help or point me in right direction to make use of DrawSVG to for QD

**Comment by cduck** [comment]
You can use an SVG path (`drawSvg.Path`) for each stroke of the drawing.  You need to loop through the stroke data to create the SVG paths.

Here is a quick demo:

```python
import drawSvg as draw

drawing_data = [
    [[251,255,242,232,237,249],
      [0,153,153,166,170,160]],
     [[253,253,248,236,232],
      [154,164,172,174,167]],
     [[2,9,32,45,59,69,66],
      [2,32,88,102,107,103,94]],
     [[0,6,11,31,58,79,86,86,69],
      [3,19,73,111,128,128,122,106,91]]
]

min_x = min(x for stroke_data in drawing_data
              for x in stroke_data[0])
max_x = max(x for stroke_data in drawing_data
              for x in stroke_data[0])
min_y = min(y for stroke_data in drawing_data
              for y in stroke_data[1])
max_y = max(y for stroke_data in drawing_data
              for y in stroke_data[1])
d = draw.Drawing(max_x-min_x+10, max_y-min_y+10,
                 origin=(min_x-5, min_y-5))

for stroke_data in drawing_data:
    x_arr = stroke_data[0]
    y_arr = stroke_data[1]
    # Create a path for each drawing stroke
    path = draw.Path(stroke_width=3, stroke='black', fill='none')
    path.M(x_arr[0], y_arr[0])  # Start of path
    for x, y in zip(x_arr[1:], y_arr[1:]):
        path.L(x, y)  # Draw a line to the next point
    d.append(path)

#d.saveSvg('drawing.svg')
d  # View in Jupyter
```

![preview](https://user-images.githubusercontent.com/2476062/58450220-80f2fc80-80fd-11e9-9500-d37f492c43ac.png)

**Comment by shubhank008** [comment]
Woot, works awesome. 
Just small problem, images are inverted (I think its reading or drawing from bottom to top or in reverse), please see attached results.

Also, is it possible to record/create a animation or gif of the drawing process ? Similar to the preview on google's page where they show a animated gif of the drawing (process)

<img width="272" alt="chrome_2019-05-28_10-10-00" src="https://user-images.githubusercontent.com/3790163/58451388-e6141580-8130-11e9-8f74-cc892112e5ec.png">
<img width="233" alt="chrome_2019-05-28_10-09-53" src="https://user-images.githubusercontent.com/3790163/58451389-e6acac00-8130-11e9-92f8-197a843b02fc.png">

**Comment by cduck** [comment]
drawSvg uses a coordinate system with the origin in the bottom left and y increasing upward.  Just negate the y-value to flip it.

Animate is a new feature in drawSvg and it can currently only make animations with a fixed frame rate.  To make animations with the animation context manager:

```python
with draw.animate_jupyter(delay=1/5) as anim:
# Or
#with draw.animate_video('drawing.gif', duration=1/5) as anim:
#with draw.animate_video('drawing.mp4', fps=5) as anim:
    # Create the drawing
    d = draw.Drawing(...)
    # ...
    anim.draw_frame(d)
```

For something like this, you can render each frame as you loop through the points:

```python
import copy
import drawSvg as draw

drawing_data = [
    [[251,255,242,232,237,249],
      [0,153,153,166,170,160]],
     [[253,253,248,236,232],
      [154,164,172,174,167]],
     [[2,9,32,45,59,69,66],
      [2,32,88,102,107,103,94]],
     [[0,6,11,31,58,79,86,86,69],
      [3,19,73,111,128,128,122,106,91]]
]

min_x = min(x for stroke_data in drawing_data
              for x in stroke_data[0])
max_x = max(x for stroke_data in drawing_data
              for x in stroke_data[0])
min_y = min(y for stroke_data in drawing_data
              for y in stroke_data[1])
max_y = max(y for stroke_data in drawing_data
              for y in stroke_data[1])
d = draw.Drawing(max_x-min_x+10, max_y-min_y+10,
                 origin=(min_x-5, -max_y-5))
# Draw a white background for the GIF
d.append(draw.Rectangle(min_x-5, -max_y-5,
                        max_x-min_x+10, max_y-min_y+10,
                        fill='white'))
d.setRenderSize(w=300)

with draw.animate_jupyter(delay=1/5) as anim:
#with draw.animate_video('drawing.gif', duration=1/5) as anim:
#with draw.animate_video('drawing.mp4', fps=5) as anim:
    for stroke_data in drawing_data:
        x_arr = stroke_data[0]
        y_arr = stroke_data[1]
        # Create a path for each drawing stroke
        path = draw.Path(stroke_width=3, stroke='black', fill='none')
        path.M(x_arr[0], -y_arr[0])  # Start of path
        d.append(path)
        for x, y in zip(x_arr[1:], y_arr[1:]):
            path.L(x, -y)  # Draw a line to the next point
            anim.draw_frame(copy.deepcopy(d))  # Render a copy of the frame
```

![drawing](https://user-images.githubusercontent.com/2476062/58452002-42614000-8105-11e9-9e0c-64cf2718cc09.gif)

**Comment by shubhank008** [comment]
Thank you for replying back so fast.
Sorry to sound complete idiot, I followed your steps and I think the image is reversed now (actually proper) but it seems to be going out of bounds/getting cut of as I think its not originating from center/top left

Here is my code

```
import copy
import drawSvg as draw

drawing_data = [((95, 79, 68, 31, 17, 9, 1, 0, 4, 54, 103, 130, 168, 190, 204, 219, 228, 222, 210, 200, 194, 197, 203, 192, 189, 190, 209, 207, 197, 179, 107, 100), (62, 50, 49, 74, 91, 113, 163, 220, 226, 249, 255, 255, 244, 231, 215, 187, 152, 151, 160, 160, 156, 140, 137, 134, 115, 109, 88, 82, 71, 63, 62, 65)), ((100, 100, 104, 110, 115, 115, 108, 107), (58, 17, 1, 0, 12, 22, 44, 64))]


min_x = min(x for stroke_data in drawing_data
              for x in stroke_data[0])
max_x = max(x for stroke_data in drawing_data
              for x in stroke_data[0])
min_y = min(y for stroke_data in drawing_data
              for y in stroke_data[1])
max_y = max(y for stroke_data in drawing_data
              for y in stroke_data[1])
d = draw.Drawing(max_x-min_x+10, max_y-min_y+10,
                 origin=(min_x-5, -min_y-5))
<img width="811" alt="chrome_2019-05-28_10-43-07" src="https://user-images.githubusercontent.com/3790163/58452556-63da2000-8135-11e9-9509-1ce715287edb.png">


d.setRenderSize(w=300)

for stroke_data in drawing_data:

    x_arr = stroke_data[0]
    y_arr = stroke_data[1]


    # Create a path for each drawing stroke
    path = draw.Path(stroke_width=3, stroke='black', fill='none')
    path.M(x_arr[0], -y_arr[0])  # Start of path
    for x, y in zip(x_arr[1:], y_arr[1:]):
        path.L(x, -y)  # Draw a line to the next point
    d.append(path)

#d.setRenderSize(w=300)
#d.saveSvg('drawing.svg')
d  # View in Jupyter
```

**Comment by shubhank008** [comment]
Nevermind, found my mistake

Was still doing
`d = draw.Drawing(max_x-min_x+10, max_y-min_y+10,
                 origin=(min_x-5, -min_y-5))`

instead of

```
d = draw.Drawing(max_x-min_x+10, max_y-min_y+10,
                 origin=(min_x-5, -max_y-5))
```

Even now I did not see and realize the mistake in code, but instead I opened a graph image on net and started to visualize the coordinates and origin and realized my mistake.
Phew, thank you for all the help. So glad that got everything working perfectly how I wanted it after 7 hours of hit and trying various libraries and methods

I might write a wrapper to create a CLI or API version of the whole method and submit a pull to both DrawSVG and google's dataset.
